sample_factory/algo/sampling/non_batched_sampling.py

```diff
@@ -494,14 +503,20 @@ def _process_policy_outputs(self, policy_id, timing):
         # a simulation step right away, without waiting for all other actions to be calculated.
         return all_actors_ready
 
-    def _process_rewards(self, rewards, env_i):
+    def _process_rewards(self, rewards, infos, env_i):
         """
         Pretty self-explanatory, here we record the episode reward and apply the optional clipping and
         scaling of rewards.
         """
         for agent_i, r in enumerate(rewards):
             self.actor_states[env_i][agent_i].last_episode_reward += r
 
+        if self.cfg.use_record_episode_statistics:
```

Comment:
Can we do all of this in one function "record_env_step" ?
Also, it looks like there is no point checking this on every step, can't we just look at info["episode"] when done=True like it's done in record_env_step() for "extra stats"?

Reply:
Also, we might need to implement the same thing in batched mode.
Look at _process_env_step there

Reply:
Will take a look at that.